src/common/libraries/RequestCallbackMessageLib.sol

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.28;
+
+import {BytesLib} from "src/misc/libraries/BytesLib.sol";
+import {CastLib} from "src/misc/libraries/CastLib.sol";
+
+enum RequestCallbackType {
```

Comment:
NIT. Not sure if the "callback" name correctly defines all types of messages. Some of the messages are not necessarily callbacks. Maybe `RequestNotification` is s wide term? 🤔

Reply:
I think callback seems quite correct here since it's the return action of some request. It is imo more than a notification.

Reply:
In terms of them being core: I think you could argue we could build an implementation of `Spoke` on a new chain (e.g. Solana) without building vaults, and in that case we wouldn't need to implement the request and request callback messages. So it is decoupled from the core in that sense.

Reply:
🤔 m but for example, in case of `hub.issueShares()`, it's not a callback right? it's just an action from hub -> spoke

Reply:
It's a second order callback from requestDeposit